File: 01/one.py
from functools import reduce


def calc(input_data):
    total = 0
    for shift in input_data:
        total += int(shift)
    return total


if __name__ == "__main__":
    with open("input.txt") as file:
        # read the input, split by lines and use it as argument to main function
        input_data = file.read().split('\n')
        output = calc(input_data)
        print(output)

File: 01/one.pl
use strict;
use warnings;

sub calc {
    my ($input_data) = @_;
    my $total = 0;
    for my $shift (@$input_data) {
        $total += $shift;
    }
    return $total;
}

open(my $file, '<', 'input.txt') or die "Could not open input.txt: $!";
# read the input and split by lines to use as the argument to the calc function
my @input_data = split(/\n/, do { local $/; <$file> });
close($file);

print calc(\@input_data), "\n";
